Update Agent Builder built-in skills reference for 9.5 (#7255)
elastic/docs-content: explore-analyze/ai-features/agent-builder/builtin-skills-reference.md

    :::{dropdown} Assigned tools
    `platform.core.execute_esql`, plus an internal {{esql}} generation tool scoped to trace data.

    :::

    **Prerequisites:** Controlled by the `agentBuilder:tracing:enabled` [advanced setting](kibana://reference/advanced-settings.md#kibana-general-settings), which is on by default.

### Cases and alerting

$$$agent-builder-cases-management-skill$$$ `cases-management` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`
:   Manages investigation and incident cases across {{elastic-sec}}, {{observability}}, and Stack Management. Covers creating, updating, searching, and enriching cases with comments, alerts, events, and observables such as indicators of compromise.

    :::{dropdown} Assigned tools
    `platform.core.cases`, `platform.core.cases.manage`, `platform.core.cases.attachments`, `platform.core.cases.observables`

    :::

$$$agent-builder-rule-management-skill$$$ `rule-management` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`

> :   Composes, discovers, and modifies alerting rules and action policies (notification policies) from within a conversation.

Reviewer: same as in https://github.com/elastic/docs-content/pull/7255/changes#r3578906285

Author: Done in the same change: removed "notification policies" from the description in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb

    :::{dropdown} Assigned tools

>     `platform.alerting.manage_rule`, plus a skill-scoped inline tool for composing and modifying action policies (notification policies).

Reviewer: I think we should remove `notification policies` to avoid confusing users. IIC, `action policies` is the correct term. 

Also, the id of the action policy tool is `platform.alerting.manage_action_policy`, in case. you want to add it the same as the rules one

Author: Added in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb. Removed "notification policies" and listed `platform.alerting.manage_action_policy` alongside the rules tool.

    :::

    :::{note}
    Without this skill, the agent can still check the status of a workflow execution and resume a paused workflow that is waiting for human input, using the [`platform.core.get_workflow_execution_status`](tools/builtin-tools-reference.md) and [`platform.core.resume_workflow_execution`](tools/builtin-tools-reference.md) tools. To trigger a specific workflow from a conversation, configure a [workflow tool](tools/workflow-tools.md) and assign it to the agent.
    :::

### Agent and skill authoring

$$$agent-builder-skill-authoring-skill$$$ `skill-authoring` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`

> :   Authors a new {{agent-builder}} skill from a chat description. Use when a user asks to create, build, generate, or design a skill, capability, or expertise area for an agent.

Reviewer 2: ```suggestion
$$$agent-builder-skill-management-skill$$$ `skill-management` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`
  :   Authors and edits {{agent-builder}} skills from a chat description. Use when a user asks to create, build, generate, design, or modify a skill, capability, or expertise area for an agent.
```

Author: Added in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb

    :::{dropdown} Assigned tools
    Internal tools to list available tools and to propose or patch a skill definition.

>     :::

Reviewer 3: @leemthompo has 9.5 been cut for release? i think the `skill_authoring_skill` is now called `skill_management_skill` https://github.com/elastic/kibana/pull/271463/changes#diff-c0bcb0f583598be2dbf7d273ffc206d7eb932385e7ce3e5945354a37686e2ea1 based on this change

Author: Good catch. 9.5 hasn't been released yet (current GA is 9.4.x). Renamed the skill to `skill-management` and updated the description to cover the view and edit flows in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb

$$$agent-builder-entity-analytics-skill$$$ `entity-analytics` {applies_to}`stack: ga 9.4+`
:   Finds and investigates security entities including hosts, users, services, and generic entities. Analyzes entity risk scores, asset criticality, and historical behavior, including signals from Security {{ml-app}} anomaly detection jobs. Use to discover risky entities or profile a specific entity by ID.

    :::{dropdown} Assigned tools
    `security.get_entity`, `security.search_entities`

    :::

>     {applies_to}`stack: ga 9.5+` From 9.5, the skill can also list watchlists to discover watchlist names and members, using the `security.list_watchlists` tool.

Reviewer 4: this skill also now also have access to `set_asset_criticality` tool
https://github.com/elastic/kibana/pull/275477

Author: thanks @tcalopes!

    **Prerequisites:** [Entity risk scoring](/solutions/security/advanced-entity-analytics/entity-risk-scoring.md) enabled and the [entity store](/solutions/security/advanced-entity-analytics/entity-store.md) populated.

    **Related skills:** [`find-security-ml-jobs`](#agent-builder-find-security-ml-jobs-skill) for deeper investigation of anomalies surfaced during entity analysis. [`manage-watchlists`](#agent-builder-manage-watchlists-skill) to create and modify watchlists.

$$$agent-builder-entity-analytics-leads-skill$$$ `entity-analytics-leads` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`

> :   Surfaces AI-generated investigation leads for security entities. Use when a user asks to review, list, show, triage, dismiss, or generate investigation leads, or wants to find proactive threat hunting opportunities surfaced from entity data.

Reviewer 5: ```suggestion
:   Surfaces AI-generated threat hunting leads for security entities. Use when a user asks to review, list, show, triage, dismiss, or generate threat hunting leads, or wants to find proactive threat hunting opportunities surfaced from entity data.
```

The "Use when a user asks..." phrasing reads like we're addressing Agent Builder (Hey Agent Builder, use this skill when a Kibana user asks this or that) rather than an actual user. I'd suggest changing it to either "Used when you ask..."  or "Use this skill to ask...".

Even more prominent on line 367:

> Use when a user asks you to walk them through...

Author: Added the threat hunting leads wording in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb. On the "Use when a user asks" phrasing, agreed it reads like guidance to the agent rather than the reader. It recurs across most entries in this reference, so I'd rather fix it consistently in a dedicated follow-up than change a single entry here. Noted to do that.

    :::{dropdown} Assigned tools
    `security.list_leads`, `security.generate_leads`, `security.dismiss_lead`

    :::

    **Prerequisites:** The `leadGenerationEnabled` {{elastic-sec}} [experimental feature flag](kibana://reference/configuration-reference/security-solution-settings.md#experimental-features) must be enabled.

> $$$agent-builder-manage-watchlists-skill$$$ `manage-watchlists` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`

Reviewer 5: ```suggestion
$$$agent-builder-manage-watchlists-skill$$$ `manage-watchlists` {applies_to}`stack: ga 9.5` {applies_to}`serverless: ga`
```

Author: Added in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb. Used the open-ended `ga 9.5+` form to match the rest of the file, and dropped the on-by-default feature-flag prerequisite.

  elasticsearch: ga
```

$$$agent-builder-search-elasticsearch-onboarding-skill$$$ `search.elasticsearch-onboarding` {applies_to}`stack: ga 9.4+`
:   Guides developers through building a complete search experience on {{es}}, from understanding requirements and designing an index mapping to generating and testing API snippets in Dev Tools. Use for end-to-end onboarding rather than a single narrow API answer.

> $$$agent-builder-search-elasticsearch-tutorial-skill$$$ `search.elasticsearch-tutorial` {applies_to}`stack: preview 9.5`

Reviewer 2: ```suggestion
$$$agent-builder-search-elasticsearch-tutorial-skill$$$ `search.elasticsearch-tutorial` {applies_to}`stack: preview 9.5` {applies_to}`serverless: preview`
```

Author: Added in https://github.com/elastic/docs-content/pull/7255/commits/16912b5e5b7bcf8d5ba89edc922185311e9b0ecb
